### PR TITLE
Steamworks mod support

### PR DESCRIPTION
**Pull request by nadult**
Adds steamworks interfaces and steam mods support. First 6 commits are simple improvements for utility libraries, etc. 7th commit adds all steam interfaces and last one integrates then into the game.

### CODE REVIEW

File: directory_path.cpp [hunk 1]
```diff
@@ -28,6 +29,15 @@ bool DirectoryPath::exists() const {
   return isDirectory(getPath());
 }
 
+bool DirectoryPath::make() {
```
Comment (miki151) [inline directory_path.cpp]: Doesn't this have the same functionality as the function below it? You only call it with 'if (!to.exists()) to.make();'

File: file_sharing.cpp [hunk 2]
```diff
@@ -334,13 +342,171 @@ static optional<FileSharing::OnlineModInfo> parseModInfo(const vector<string>& f
   return none;
 }
 
+static string shortDescription(string text, int max_lines = 3) {
```
Comment (miki151) [inline file_sharing.cpp]: Maybe rename the function to what it actually does, which is returning the first max_lines lines from text?

File: directory_path.cpp (hunk 1 above)
Comment (nadult) [reply directory_path.cpp]: Yes, I didn't notice it :)

File: file_sharing.cpp (hunk 2 above)
Comment (nadult) [reply file_sharing.cpp]: Sure

**Pull request merged**